### ISSUE TITLE
Default valueFormatter for a given cellDataType is not invoked on parent rows in a tree grid

### ISSUE THREAD

**Issue by OysteinAmundsen** [ag-grid/ag-grid#8639]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Providing a Reproducible Scenario**
[Stackblitz example on bug](https://stackblitz.com/edit/stackblitz-starters-hvnbqe?file=tsconfig.app.json,environment%2Fenvironment.ts,src%2Frenderers%2Fdate-cell-renderer.component.ts)

**Current behavior**
As this demonstrates, the renderer is unable to render the given date on non-leaf rows when running AG-Grid as tree. The same date is given as millis on both the row group and the child row, but only the child row displays it as a date.

I've found when debugging that the default valueFormatter for cellDataType 'millis' is not present in the ICellRendererParams given to the custom renderer component on non-leaf rows. The function is only ever invoked on leaf rows.

**Expected behavior**
I would expect the parent rows to invoke the same valueFormatter as the leaf nodes does.

**Please tell us about your environment:**
Windows 11, VSCode, bun, ESBuild + Vite

* **AG Grid version:** X.X.X
"@ag-grid-community/angular": "^32.1.0",

* **Browser:**
Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/128.0.0.0 Safari/537.36

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
"typescript": "5.5.4"

**Comment by OysteinAmundsen** [comment]
This has been marked as [#AG-12788 - When using cell data types, the value formatter does not run correctly for group rows, including when using tree data](https://www.ag-grid.com/pipeline/)

**Comment by AG-Zoheil** [comment]
Thank you for your comment.

**AG-12788** was released as part of version `32.2.0`.
